gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/writer/commands/BaseJdbcBufferedInserter.java

```diff
@@ -80,6 +85,8 @@ public BaseJdbcBufferedInserter(State state, Connection conn) {
     this.retryer = RetryerBuilder.<Boolean> newBuilder().retryIfException()
         .withWaitStrategy(WaitStrategies.exponentialWait(1000, maxWait, TimeUnit.SECONDS))
         .withStopStrategy(StopStrategies.stopAfterAttempt(maxAttempts)).build();
+
+    this.replaceExistingValues = state.getPropAsBoolean(ConfigurationKeys.ALLOW_DATA_OVERWRITE);
```

Comment:
I haven't added the "fail fast" logic in the other child classes that do not have any functionality implemented for this. If the config is true should the job fail in the constructor or at the time of writing `(initialize batch)` or should I add documentation that the logic does not exist and allow the job to succeed?